### PR TITLE
Added universe tests and minor fixes

### PR DESCRIPTION
**Pull request by pradyuman-verma**
- [x] Added test for universe finance
- [x] Added get position functions and decimals function 
- [x] Deploy and comment address

### CODE REVIEW

**Comment by pradyuman-verma** [comment]
test working properly: 
image
image

File: contracts/protocols/mainnet/universeFinance/main.sol [hunk 1]
```diff
@@ -19,43 +18,58 @@ abstract contract UniverseFinanceResolver is Helpers {
...
     }
 
+    /**
+     * @notice get universe vaults data
+     * @param universeVaults the Universe Vault Address array
+     * @return array of [token0Address, token1Address, vaultMaxToken0Amount, vaultMaxToken1Amount, 
+        maxSingleDepositFofToken0,maxSingleDepositFofToken1, totalToken0Amount, totalTotal1Amount, 
+        utilizationOfToken0, utilizationOfToken1]
+     */
+    function getVaultData(address[] memory universeVaults) external view returns (VaultData[] memory) {
```
Comment (KaymasJain) [inline contracts/protocols/mainnet/universeFinance/main.sol]: Need to add one more function like this where we are not sending any vaults and it's directly fetching it from `getAllVault()`.

File: contracts/protocols/mainnet/universeFinance/main.sol [hunk 2]
```diff
@@ -71,10 +85,37 @@ abstract contract UniverseFinanceResolver is Helpers {
...
+    function decimals(address vault) external view returns (uint8, uint8) {
+        return _decimals(vault);
+    }
+
+    /**
+     * @notice get token decimals of a vault
+     * @param universeVault the vault's address
+     * @param user the user's address
+     */
+    function position(address universeVault, address user)
```
Comment (KaymasJain) [inline contracts/protocols/mainnet/universeFinance/main.sol]: Add function to get multiple user Universe Vault position for a user.

**Comment by KaymasJain** [comment]
Add 2 more functions as:
- Takes users address & Universe vaults addresses and returns vaults data & users position. (would be mix of 2 current functions).
- Takes users address and returns vaults data & users position. (would be mix of 2 current functions). (fetches all the vaults by `getAllVault`).

File: contracts/protocols/mainnet/universeFinance/main.sol (hunk 2 above)
Comment (pradyuman-verma) [reply contracts/protocols/mainnet/universeFinance/main.sol]: okay

**Comment by pradyuman-verma** [comment]
test: 
image
image

**Comment by pradyuman-verma** [comment]
~TODO: create a generalized `position` function which returns all the required user info in one call!~

**Review by KaymasJain — APPROVED** [review]

**Comment by pradyuman-verma** [comment]
Deployed on mainnet: [0x696360Ad22eBc119Da7004Da0DD948aaBAD22948](https://etherscan.io/address/0x696360Ad22eBc119Da7004Da0DD948aaBAD22948)

**Review by KaymasJain — APPROVED** [review]

**Pull request merged**